subsys/bluetooth/services/hrs.c

```diff
@@ -91,6 +97,17 @@ static int hrs_init(void)
 	return 0;
 }
 
+int bt_hrs_cb_register(struct bt_hrs_cb *cb)
+{
+	if (cb == NULL) {
+		return -EINVAL;
+	}
+
+	hrs_cb = cb;
```

Comment:
I would like us to display a helpful error message if the user hits the limitation that it's not possible to register multiple listeners.

```suggestion
	if (!atomic_ptr_cas((atomic_ptr_t)&hrs_cb, NULL, (atomic_ptr_val_t)cb)) {
		LOG_ERR("Not implemented: Registering multiple rate service callbacks");
		return -ENOSYS;
	}
```